api/pkg/cli/cmd/install/install.go

```diff
@@ -0,0 +1,212 @@
+// Copyright © 2020 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package install
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/hub/api/pkg/cli/app"
+	"github.com/tektoncd/hub/api/pkg/cli/flag"
+	"github.com/tektoncd/hub/api/pkg/cli/hub"
+	"github.com/tektoncd/hub/api/pkg/cli/kube"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type options struct {
+	kube.Client
+	cli      app.CLI
+	from     string
+	version  string
+	kind     string
+	args     []string
+	upgrade  bool
+	resource *hub.ResourceResult
+}
+
+var cmdExamples string = `
+Install a %S of name 'foo':
+
+    tkn hub install %s foo
+
+or
+
+Install a %S of name 'foo' of version '0.3' from Catalog 'Tekton':
+
+    tkn hub install %s foo --version 0.3 --from tekton
+`
+
+func Command(cli app.CLI) *cobra.Command {
+
+	opts := &options{cli: cli, Client: &kube.ClientParam{}}
```

Comment:
how about `kc: kube.NewClient()` instead of `Client: &kube.ClientParam{}`. We can keep the actual client internal to `kube` package.

Reply:
Yeah Thanks :)